annotationframeworkclient/frameworkclient.py

```diff
@@ -0,0 +1,246 @@
+
+    Parameters
+    ----------
+    dataset_name : str, optional
+        Dataset name for the services. Almost all services need this and will not work if it is not passed.
+    server_address : str or None
+        URL of the framework server. If None, chooses the default server www.dynamicannotationframework.com.
+        Optional, defaults to None.
+    auth_token_file : str or None
+        Path to a json file containing the auth token. If None, uses the default location. See Auth client documentation.
+        Optional, defaults to None.
+    auth_token_key : str
+        Dictionary key for the token in the the JSON file.
+        Optional, default is 'token'.
+    auth_token : str or None
+        Direct entry of an auth token. If None, uses the file arguments to find the token.
+        Optional, default is None.
+    """
+
+    def __init__(
+        self,
+        dataset_name=None,
+        server_address=None,
+        auth_token_file=default_token_file,
+        auth_token_key="token",
+        auth_token=None,
+    ):
+        self._dataset_name = dataset_name
+        if server_address is None:
+            server_address = default_server_address
+        self._server_address = server_address
+        self._auth_config = (
+            auth_token_file,
+            auth_token_key,
+            auth_token,
+            server_address,
+        )
+
+        self._auth = None
+        self._info = None
+        self._state = None
+        self._schema = None
+        self._chunkedgraph = None
+        self._annotation = None
+        self._lookup = None
+
+    def change_auth(self, auth_token_file=None, auth_token_key=None, auth_token=None):
+        """Change the authentication token and reset services.
+
+        Parameters
+        ----------
+        auth_token_file : str, optional
+            New auth token json file path, by default None, which defaults to the existing state.
+        auth_token_key : str, optional
+            New dictionary key under which the token is stored in the json file, by default None,
+            which defaults to the existing state.
+        auth_token : str, optional
+            Direct entry of a new token, by default None.
+        """
+        if auth_token_file is None:
+            auth_token_file = self._auth_config[0]
+        if auth_token_key is None:
+            auth_token_key = self._auth_config[1]
+
+        self._auth_config = (
+            auth_token_file,
+            auth_token_key,
+            auth_token,
+            self._server_address,
+        )
+        self._reset_services()
+
+    def _reset_services(self):
+        self._auth = None
+        self._info = None
+        self._state = None
+        self._schema = None
+        self._chunkedgraph = None
+        self._annotation = None
+        self._imagery = None
+        self._lookup = None
+
+    @property
+    def dataset_name(self):
+        return self._dataset_name
+
+    @property
+    def server_address(self):
```

Comment:
i'm not sure this makes sense in a world where auth_client and info service will live on different servers than the chunkedgraph and annotation engine